File: dashboard_webserver/bindings.ts
import {
  WeilWallet,
  Schema,
  Contract,
  ContractFactory,
  parseSchema,
  parseExecutionResult,
  WeilWalletConnection,
  Option,
} from '@weilliptic/weil-sdk';


interface DashboardConfig {
  name: string;
}
const dashboardConfigSchema = 
  Schema.object({
    name: Schema.string,
  });



interface Alert {
  id: string;
  alert_type: string;
  severity: string;
  risk_score: number;
  entity_id: string;
  symbol: string;
  description: string;
  workflow_id: string;
  timestamp: bigint;
}
const alertSchema = 
  Schema.object({
    id: Schema.string,
    alert_type: Schema.string,
    severity: Schema.string,
    risk_score: Schema.u32,
    entity_id: Schema.string,
    symbol: Schema.string,
    description: Schema.string,
    workflow_id: Schema.string,
    timestamp: Schema.u64,
  });



interface WorkflowExecution {
  id: string;
  workflow_type: string;
  trigger: string;
  steps_completed: number;
  total_steps: number;
  status: string;
  started_at: bigint;
  completed_at: bigint;
  result_summary: string;
}
const workflowExecutionSchema = 
  Schema.object({
    id: Schema.string,
    workflow_type: Schema.string,
    trigger: Schema.string,
    steps_completed: Schema.u32,
    total_steps: Schema.u32,
    status: Schema.string,
    started_at: Schema.u64,
    completed_at: Schema.u64,
    result_summary: Schema.string,
  });



interface CaseRecord {
  case_id: string;
  case_type: string;
  status: string;
  priority: string;
  subject_entity: string;
  symbol: string;
  risk_score: number;
  assigned_to: string;
  created_at: bigint;
  updated_at: bigint;
  summary: string;
}
const caseRecordSchema = 
  Schema.object({
    case_id: Schema.string,
    case_type: Schema.string,
    status: Schema.string,
    priority: Schema.string,
    subject_entity: Schema.string,
    symbol: Schema.string,
    risk_score: Schema.u32,
    assigned_to: Schema.string,
    created_at: Schema.u64,
    updated_at: Schema.u64,
    summary: Schema.string,
  });



interface SurveillanceStats {
  total_alerts_today: number;
  total_workflows_today: number;
  open_cases: number;
  high_risk_entities: number;
  compliance_score: number;
}
const surveillanceStatsSchema = 
  Schema.object({
    total_alerts_today: Schema.u32,
    total_workflows_today: Schema.u32,
    open_cases: Schema.u32,
    high_risk_entities: Schema.u32,
    compliance_score: Schema.u32,
  });



interface RiskEntity {
  entity_id: string;
  entity_name: string;
  risk_score: number;
  alert_count: number;
  last_alert_at: bigint;
}
const riskEntitySchema = 
  Schema.object({
    entity_id: Schema.string,
    entity_name: Schema.string,
    risk_score: Schema.u32,
    alert_count: Schema.u32,
    last_alert_at: Schema.u64,
  });



interface Trade {
  trade_id: string;
  symbol: string;
  account_id: string;
  trade_type: string;
  quantity: bigint;
  price: string;
  value: string;
  exchange: string;
  segment: string;
  timestamp: bigint;
  order_id: string;
}
const tradeSchema = 
  Schema.object({
    trade_id: Schema.string,
    symbol: Schema.string,
    account_id: Schema.string,
    trade_type: Schema.string,
    quantity: Schema.u64,
    price: Schema.string,
    value: Schema.string,
    exchange: Schema.string,
    segment: Schema.string,
    timestamp: Schema.u64,
    order_id: Schema.string,
  });



interface Entity {
  entity_id: string;
  entity_type: string;
  name: string;
  pan_number: string;
  registration_id: string;
}
const entitySchema = 
  Schema.object({
    entity_id: Schema.string,
    entity_type: Schema.string,
    name: Schema.string,
    pan_number: Schema.string,
    registration_id: Schema.string,
  });



interface Relationship {
  source_entity_id: string;
  target_entity_id: string;
  relationship_type: string;
  relationship_detail: string;
  strength: number;
  verified: boolean;
}
const relationshipSchema = 
  Schema.object({
    source_entity_id: Schema.string,
    target_entity_id: Schema.string,
    relationship_type: Schema.string,
    relationship_detail: Schema.string,
    strength: Schema.u32,
    verified: Schema.bool,
  });



interface InsiderStatus {
  entity_id: string;
  company_symbol: string;
  is_insider: boolean;
  insider_type: string;
  designation: string;
  window_status: string;
}
const insiderStatusSchema = 
  Schema.object({
    entity_id: Schema.string,
    company_symbol: Schema.string,
    is_insider: Schema.bool,
    insider_type: Schema.string,
    designation: Schema.string,
    window_status: Schema.string,
  });



interface ReportResult {
  report_id: string;
  report_type: string;
  storage_path: string;
  download_url: string;
  expires_at: bigint;
  risk_score: number;
  success: boolean;
  error: string;
}
const reportResultSchema = 
  Schema.object({
    report_id: Schema.string,
    report_type: Schema.string,
    storage_path: Schema.string,
    download_url: Schema.string,
    expires_at: Schema.u64,
    risk_score: Schema.u32,
    success: Schema.bool,
    error: Schema.string,
  });



interface UPSIRecord {
  upsi_id: string;
  company_symbol: string;
  upsi_type: string;
  description: string;
  nature: string;
  created_date: bigint;
  public_date: bigint;
  is_public: boolean;
}
const upsiRecordSchema = 
  Schema.object({
    upsi_id: Schema.string,
    company_symbol: Schema.string,
    upsi_type: Schema.string,
    description: Schema.string,
    nature: Schema.string,
    created_date: Schema.u64,
    public_date: Schema.u64,
    is_public: Schema.bool,
  });



interface TradingWindowStatus {
  company_symbol: string;
  window_status: string;
  closure_reason: string;
  closure_start: bigint;
  expected_opening: bigint;
}
const tradingWindowStatusSchema = 
  Schema.object({
    company_symbol: Schema.string,
    window_status: Schema.string,
    closure_reason: Schema.string,
    closure_start: Schema.u64,
    expected_opening: Schema.u64,
  });



interface TradeAnalysis {
  symbol: string;
  total_volume: bigint;
  avg_price: string;
  high_price: string;
  low_price: string;
  buy_volume: bigint;
  sell_volume: bigint;
  trade_count: number;
  concentration_ratio: string;
}
const tradeAnalysisSchema = 
  Schema.object({
    symbol: Schema.string,
    total_volume: Schema.u64,
    avg_price: Schema.string,
    high_price: Schema.string,
    low_price: Schema.string,
    buy_volume: Schema.u64,
    sell_volume: Schema.u64,
    trade_count: Schema.u32,
    concentration_ratio: Schema.string,
  });



export const DashboardWebserver = ((wallet: WeilWallet | WeilWalletConnection, contractAddress: string) => ({
  ping: async () => {

    const args = {}
    const result = await wallet.contracts.execute(
      contractAddress,
      "ping",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },
  push_alert: async (alert: Alert) => {

    const args = parseSchema(
      Schema.args({
        alert: alertSchema,
      }),
      { alert }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "push_alert",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },
  log_workflow_start: async (workflow_id: string, workflow_type: string, trigger: string, total_steps: number) => {

    const args = parseSchema(
      Schema.args({
        workflow_id: Schema.string,
        workflow_type: Schema.string,
        trigger: Schema.string,
        total_steps: Schema.u32,
      }),
      { workflow_id, workflow_type, trigger, total_steps }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "log_workflow_start",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },
  update_workflow_progress: async (workflow_id: string, steps_completed: number, status: string, result_summary: string) => {

    const args = parseSchema(
      Schema.args({
        workflow_id: Schema.string,
        steps_completed: Schema.u32,
        status: Schema.string,
        result_summary: Schema.string,
      }),
      { workflow_id, steps_completed, status, result_summary }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "update_workflow_progress",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },
  upsert_case: async (case_record: CaseRecord) => {

    const args = parseSchema(
      Schema.args({
        case_record: caseRecordSchema,
      }),
      { case_record }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "upsert_case",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },
  register_risk_entity: async (entity: RiskEntity) => {

    const args = parseSchema(
      Schema.args({
        entity: riskEntitySchema,
      }),
      { entity }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "register_risk_entity",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },
  get_live_alerts: async (severity_filter: Option<string>, limit: Option<number>) => {

    const args = parseSchema(
      Schema.args({
        severity_filter: Schema.option(Schema.string),
        limit: Schema.option(Schema.u32),
      }),
      { severity_filter, limit }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_live_alerts",
      args,
    );

    return parseExecutionResult(result, Schema.array(alertSchema));
  },
  get_workflow_history: async (workflow_type: Option<string>, limit: Option<number>) => {

    const args = parseSchema(
      Schema.args({
        workflow_type: Schema.option(Schema.string),
        limit: Schema.option(Schema.u32),
      }),
      { workflow_type, limit }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_workflow_history",
      args,
    );

    return parseExecutionResult(result, Schema.array(workflowExecutionSchema));
  },
  get_cases_by_status: async (status: Option<string>, limit: Option<number>) => {

    const args = parseSchema(
      Schema.args({
        status: Schema.option(Schema.string),
        limit: Schema.option(Schema.u32),
      }),
      { status, limit }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_cases_by_status",
      args,
    );

    return parseExecutionResult(result, Schema.array(caseRecordSchema));
  },
  get_stats: async () => {

    const args = {}
    const result = await wallet.contracts.execute(
      contractAddress,
      "get_stats",
      args,
    );

    return parseExecutionResult(result, surveillanceStatsSchema);
  },
  get_high_risk_entities: async (min_risk_score: Option<number>, limit: Option<number>) => {

    const args = parseSchema(
      Schema.args({
        min_risk_score: Schema.option(Schema.u32),
        limit: Schema.option(Schema.u32),
      }),
      { min_risk_score, limit }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_high_risk_entities",
      args,
    );

    return parseExecutionResult(result, Schema.array(riskEntitySchema));
  },
  get_case_details: async (case_id: string) => {

    const args = parseSchema(
      Schema.args({
        case_id: Schema.string,
      }),
      { case_id }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_case_details",
      args,
    );

    return parseExecutionResult(result, caseRecordSchema);
  },
  get_entity_alerts: async (entity_id: string, limit: Option<number>) => {

    const args = parseSchema(
      Schema.args({
        entity_id: Schema.string,
        limit: Schema.option(Schema.u32),
      }),
      { entity_id, limit }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_entity_alerts",
      args,
    );

    return parseExecutionResult(result, Schema.array(alertSchema));
  },
  get_trades_proxy: async (symbol: string, limit: Option<number>) => {

    const args = parseSchema(
      Schema.args({
        symbol: Schema.string,
        limit: Schema.option(Schema.u32),
      }),
      { symbol, limit }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_trades_proxy",
      args,
    );

    return parseExecutionResult(result, Schema.array(tradeSchema));
  },
  search_entities_proxy: async (search_query: string) => {

    const args = parseSchema(
      Schema.args({
        search_query: Schema.string,
      }),
      { search_query }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "search_entities_proxy",
      args,
    );

    return parseExecutionResult(result, Schema.array(entitySchema));
  },
  get_relationships_proxy: async (entity_id: string) => {

    const args = parseSchema(
      Schema.args({
        entity_id: Schema.string,
      }),
      { entity_id }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_relationships_proxy",
      args,
    );

    return parseExecutionResult(result, Schema.array(relationshipSchema));
  },
  check_insider_proxy: async (entity_id: string, company_symbol: string) => {

    const args = parseSchema(
      Schema.args({
        entity_id: Schema.string,
        company_symbol: Schema.string,
      }),
      { entity_id, company_symbol }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "check_insider_proxy",
      args,
    );

    return parseExecutionResult(result, insiderStatusSchema);
  },
  get_active_upsi_proxy: async (company_symbol: string) => {

    const args = parseSchema(
      Schema.args({
        company_symbol: Schema.string,
      }),
      { company_symbol }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_active_upsi_proxy",
      args,
    );

    return parseExecutionResult(result, Schema.array(upsiRecordSchema));
  },
  get_trading_window_proxy: async (company_symbol: string) => {

    const args = parseSchema(
      Schema.args({
        company_symbol: Schema.string,
      }),
      { company_symbol }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "get_trading_window_proxy",
      args,
    );

    return parseExecutionResult(result, tradingWindowStatusSchema);
  },
  analyze_volume_proxy: async (symbol: string) => {

    const args = parseSchema(
      Schema.args({
        symbol: Schema.string,
      }),
      { symbol }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "analyze_volume_proxy",
      args,
    );

    return parseExecutionResult(result, tradeAnalysisSchema);
  },
  generate_report_proxy: async (report_type: string, params: string) => {

    const args = parseSchema(
      Schema.args({
        report_type: Schema.string,
        params: Schema.string,
      }),
      { report_type, params }
    );

    const result = await wallet.contracts.execute(
      contractAddress,
      "generate_report_proxy",
      args,
    );

    return parseExecutionResult(result, reportResultSchema);
  },
  get_tools: async () => {

    const args = {}
    const result = await wallet.contracts.execute(
      contractAddress,
      "get_tools",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },
  get_prompts: async () => {

    const args = {}
    const result = await wallet.contracts.execute(
      contractAddress,
      "get_prompts",
      args,
    );

    return parseExecutionResult(result, Schema.string);
  },

} satisfies Contract)) satisfies ContractFactory;

File: dashboard_webserver/bindings.rs

use serde::{Deserialize, Serialize};
use weil_macros::{constructor, mutate, query, secured, smart_contract, WeilType};
use weil_rs::collections::{streaming::ByteStream, plottable::Plottable};
use weil_rs::config::Secrets;
use weil_rs::webserver::WebServer;


#[derive(Debug, Serialize, Deserialize, WeilType, Default)]
pub struct DashboardConfig {
    pub name: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Alert {
    pub id: String,
    pub alert_type: String,
    pub severity: String,
    pub risk_score: u32,
    pub entity_id: String,
    pub symbol: String,
    pub description: String,
    pub workflow_id: String,
    pub timestamp: u64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct WorkflowExecution {
    pub id: String,
    pub workflow_type: String,
    pub trigger: String,
    pub steps_completed: u32,
    pub total_steps: u32,
    pub status: String,
    pub started_at: u64,
    pub completed_at: u64,
    pub result_summary: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CaseRecord {
    pub case_id: String,
    pub case_type: String,
    pub status: String,
    pub priority: String,
    pub subject_entity: String,
    pub symbol: String,
    pub risk_score: u32,
    pub assigned_to: String,
    pub created_at: u64,
    pub updated_at: u64,
    pub summary: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SurveillanceStats {
    pub total_alerts_today: u32,
    pub total_workflows_today: u32,
    pub open_cases: u32,
    pub high_risk_entities: u32,
    pub compliance_score: u32,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct RiskEntity {
    pub entity_id: String,
    pub entity_name: String,
    pub risk_score: u32,
    pub alert_count: u32,
    pub last_alert_at: u64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct HistoryEntry {
    pub id: String,
    pub timestamp: u64,
    pub source_mcp: String,
    pub method_name: String,
    pub params: String,
    pub result_summary: String,
    pub status: String,
    pub entity_id: String,
    pub symbol: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Trade {
    pub trade_id: String,
    pub symbol: String,
    pub account_id: String,
    pub trade_type: String,
    pub quantity: u64,
    pub price: String,
    pub value: String,
    pub exchange: String,
    pub segment: String,
    pub timestamp: u64,
    pub order_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Entity {
    pub entity_id: String,
    pub entity_type: String,
    pub name: String,
    pub pan_number: String,
    pub registration_id: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Relationship {
    pub source_entity_id: String,
    pub target_entity_id: String,
    pub relationship_type: String,
    pub relationship_detail: String,
    pub strength: u32,
    pub verified: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct InsiderStatus {
    pub entity_id: String,
    pub company_symbol: String,
    pub is_insider: bool,
    pub insider_type: String,
    pub designation: String,
    pub window_status: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ReportResult {
    pub report_id: String,
    pub report_type: String,
    pub storage_path: String,
    pub download_url: String,
    pub expires_at: u64,
    pub risk_score: u32,
    pub success: bool,
    pub error: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UPSIRecord {
    pub upsi_id: String,
    pub company_symbol: String,
    pub upsi_type: String,
    pub description: String,
    pub nature: String,
    pub created_date: u64,
    pub public_date: u64,
    pub is_public: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TradingWindowStatus {
    pub company_symbol: String,
    pub window_status: String,
    pub closure_reason: String,
    pub closure_start: u64,
    pub expected_opening: u64,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TradeAnalysis {
    pub symbol: String,
    pub total_volume: u64,
    pub avg_price: String,
    pub high_price: String,
    pub low_price: String,
    pub buy_volume: u64,
    pub sell_volume: u64,
    pub trade_count: u32,
    pub concentration_ratio: String,
}

trait DashboardWebserver {
    fn new() -> Result<Self, String>
    where
        Self: Sized;
    async fn ping(&self) -> String;
    async fn push_alert(&mut self, alert: Alert) -> Result<String, String>;
    async fn log_workflow_start(&mut self, workflow_id: String, workflow_type: String, trigger: String, total_steps: u32) -> Result<String, String>;
    async fn update_workflow_progress(&mut self, workflow_id: String, steps_completed: u32, status: String, result_summary: String) -> Result<String, String>;
    async fn upsert_case(&mut self, case_record: CaseRecord) -> Result<String, String>;
    async fn register_risk_entity(&mut self, entity: RiskEntity) -> Result<String, String>;
    async fn push_history(&mut self, entry: HistoryEntry) -> Result<String, String>;
    async fn get_history(&self, source_mcp: Option<String>, limit: Option<u32>) -> Result<Vec<HistoryEntry>, String>;
    async fn get_live_alerts(&self, severity_filter: Option<String>, limit: Option<u32>) -> Result<Vec<Alert>, String>;
    async fn get_workflow_history(&self, workflow_type: Option<String>, limit: Option<u32>) -> Result<Vec<WorkflowExecution>, String>;
    async fn get_cases_by_status(&self, status: Option<String>, limit: Option<u32>) -> Result<Vec<CaseRecord>, String>;
    async fn get_stats(&self) -> Result<SurveillanceStats, String>;
    async fn get_high_risk_entities(&self, min_risk_score: Option<u32>, limit: Option<u32>) -> Result<Vec<RiskEntity>, String>;
    async fn get_case_details(&self, case_id: String) -> Result<CaseRecord, String>;
    async fn get_entity_alerts(&self, entity_id: String, limit: Option<u32>) -> Result<Vec<Alert>, String>;
    async fn get_trades_proxy(&mut self, symbol: String, limit: Option<u32>) -> Result<Vec<Trade>, String>;
    async fn search_entities_proxy(&mut self, search_query: String) -> Result<Vec<Entity>, String>;
    async fn get_relationships_proxy(&mut self, entity_id: String) -> Result<Vec<Relationship>, String>;
    async fn check_insider_proxy(&mut self, entity_id: String, company_symbol: String) -> Result<InsiderStatus, String>;
    async fn get_active_upsi_proxy(&mut self, company_symbol: String) -> Result<Vec<UPSIRecord>, String>;
    async fn get_trading_window_proxy(&mut self, company_symbol: String) -> Result<TradingWindowStatus, String>;
    async fn analyze_volume_proxy(&mut self, symbol: String) -> Result<TradeAnalysis, String>;
    async fn generate_report_proxy(&mut self, report_type: String, params: String) -> Result<ReportResult, String>;
    async fn get_tools(&self) -> String;
    async fn get_prompts(&self) -> String;

    // webserver specific functions
    fn start_file_upload(&mut self, path: String, total_chunks: u32) -> Result<(), String>;
    fn add_path_content(
        &mut self,
        path: String,
        chunk: Vec<u8>,
        index: u32,
    ) -> Result<(), String>;
    fn finish_upload(&mut self, path: String, size_bytes: u32) -> Result<(), String>;
    fn total_chunks(&self, path: String) -> Result<u32, String>;
    fn http_content(
        &self,
        path: String,
        index: u32,
        method: String,
    ) -> (u16, std::collections::HashMap<String, String>, Vec<u8>);
    fn size_bytes(&self, path: String) -> Result<u32, String>;
    fn get_chunk_size(&self) -> u32;
}

#[derive(Serialize, Deserialize, WeilType)]
pub struct DashboardWebserverContractState {
    // define your contract state here!
    secrets: Secrets<DashboardConfig>,
    server: WebServer,
}

#[smart_contract]
impl DashboardWebserver for DashboardWebserverContractState {
    #[constructor]
    fn new() -> Result<Self, String>
    where
        Self: Sized,
    {
        unimplemented!();
    }


    #[query]
    async fn ping(&self) -> String {
        unimplemented!();
    }

    #[mutate]
    async fn push_alert(&mut self, alert: Alert) -> Result<String, String> {
        unimplemented!();
    }

    #[mutate]
    async fn log_workflow_start(&mut self, workflow_id: String, workflow_type: String, trigger: String, total_steps: u32) -> Result<String, String> {
        unimplemented!();
    }

    #[mutate]
    async fn update_workflow_progress(&mut self, workflow_id: String, steps_completed: u32, status: String, result_summary: String) -> Result<String, String> {
        unimplemented!();
    }

    #[mutate]
    async fn upsert_case(&mut self, case_record: CaseRecord) -> Result<String, String> {
        unimplemented!();
    }

    #[mutate]
    async fn register_risk_entity(&mut self, entity: RiskEntity) -> Result<String, String> {
        unimplemented!();
    }

    #[mutate]
    async fn push_history(&mut self, entry: HistoryEntry) -> Result<String, String> {
        unimplemented!();
    }

    #[query]
    async fn get_history(&self, source_mcp: Option<String>, limit: Option<u32>) -> Result<Vec<HistoryEntry>, String> {
        unimplemented!();
    }

    #[query]
    async fn get_live_alerts(&self, severity_filter: Option<String>, limit: Option<u32>) -> Result<Vec<Alert>, String> {
        unimplemented!();
    }

    #[query]
    async fn get_workflow_history(&self, workflow_type: Option<String>, limit: Option<u32>) -> Result<Vec<WorkflowExecution>, String> {
        unimplemented!();
    }

    #[query]
    async fn get_cases_by_status(&self, status: Option<String>, limit: Option<u32>) -> Result<Vec<CaseRecord>, String> {
        unimplemented!();
    }

    #[query]
    async fn get_stats(&self) -> Result<SurveillanceStats, String> {
        unimplemented!();
    }

    #[query]
    async fn get_high_risk_entities(&self, min_risk_score: Option<u32>, limit: Option<u32>) -> Result<Vec<RiskEntity>, String> {
        unimplemented!();
    }

    #[query]
    async fn get_case_details(&self, case_id: String) -> Result<CaseRecord, String> {
        unimplemented!();
    }

    #[query]
    async fn get_entity_alerts(&self, entity_id: String, limit: Option<u32>) -> Result<Vec<Alert>, String> {
        unimplemented!();
    }

    #[mutate]
    async fn get_trades_proxy(&mut self, symbol: String, limit: Option<u32>) -> Result<Vec<Trade>, String> {
        unimplemented!();
    }

    #[mutate]
    async fn search_entities_proxy(&mut self, search_query: String) -> Result<Vec<Entity>, String> {
        unimplemented!();
    }

    #[mutate]
    async fn get_relationships_proxy(&mut self, entity_id: String) -> Result<Vec<Relationship>, String> {
        unimplemented!();
    }

    #[mutate]
    async fn check_insider_proxy(&mut self, entity_id: String, company_symbol: String) -> Result<InsiderStatus, String> {
        unimplemented!();
    }

    #[mutate]
    async fn get_active_upsi_proxy(&mut self, company_symbol: String) -> Result<Vec<UPSIRecord>, String> {
        unimplemented!();
    }

    #[mutate]
    async fn get_trading_window_proxy(&mut self, company_symbol: String) -> Result<TradingWindowStatus, String> {
        unimplemented!();
    }

    #[mutate]
    async fn analyze_volume_proxy(&mut self, symbol: String) -> Result<TradeAnalysis, String> {
        unimplemented!();
    }

    #[mutate]
    async fn generate_report_proxy(&mut self, report_type: String, params: String) -> Result<ReportResult, String> {
        unimplemented!();
    }

    #[query]
    async fn get_tools(&self) -> String {
        unimplemented!();
    }

    #[query]
    async fn get_prompts(&self) -> String {
        unimplemented!();
    }

    #[mutate]
    fn start_file_upload(&mut self, path: String, total_chunks: u32) -> Result<(), String> {
        self.server.start_file_upload(path, total_chunks)
    }

    #[query]
    fn total_chunks(&self, path: String) -> Result<u32, String> {
        self.server.total_chunks(path)
    }

    #[mutate]
    fn add_path_content(
        &mut self,
        path: String,
        chunk: Vec<u8>,
        index: u32,
    ) -> Result<(), String> {
        self.server.add_path_content(path, chunk, index)
    }

    #[mutate]
    fn finish_upload(&mut self, path: String, size_bytes: u32) -> Result<(), String> {
        self.server.finish_upload(path, size_bytes)
    }

    #[query]
    fn http_content(
        &self,
        path: String,
        index: u32,
        method: String,
    ) -> (u16, std::collections::HashMap<String, String>, Vec<u8>) {
        self.server.http_content(path, index, method)
    }

    #[query]
    fn size_bytes(&self, path: String) -> Result<u32, String> {
        self.server.size_bytes(path)
    }

    #[query]
    fn get_chunk_size(&self) -> u32 {
        self.server.get_chunk_size()
    }
}

